.githooks/pre-commit

```diff
@@ -57,4 +57,6 @@ done
 # If there are whitespace errors, print the offending file names and fail.
 git diff-index --check --cached $against --
 
+npm shrinkwrap --dev
```

Comment:
Oh, what writes out the JSON file then? I might be confused by the tools. I'm more than happy for something that fails if you haven't updated package.json.